### PR TITLE
Support binary predicate nullable stats 

### PR DESCRIPTION
**Pull request by HangyuanLiu**
#1100

### CODE REVIEW

**Comment by Seaven** [comment]
Add a ut plz. BTW, you modify nullable statsitic but tpch cost plan not change?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java [hunk 1]
```diff
@@ -143,7 +188,8 @@ public static Statistics estimateColumnNotEqualToColumn(
         if (leftColumn.isUnknown() || rightColumn.isUnknown()) {
             rowCount = rowCount * 0.8;
         } else {
-            rowCount = rowCount * (1.0 - selectivity);
+            rowCount = rowCount * (1.0 - selectivity)
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java:191]: Is it estimate too small?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java [hunk 2]
```diff
@@ -119,15 +125,54 @@ public static Statistics estimateColumnToColumnComparison(
...
+        ColumnStatistic newLeftStatistic;
+        ColumnStatistic newRightStatistic;
+        if (!isEqualForNull) {
+            newEstimateColumnStatistics.setNullsFraction(0);
+            newLeftStatistic = newEstimateColumnStatistics
+                    .setAverageRowSize(leftColumnStatistic.getAverageRowSize()).build();
+            newRightStatistic = newEstimateColumnStatistics
+                    .setAverageRowSize(rightColumnStatistic.getAverageRowSize()).build();
+        } else {
+            newLeftStatistic = newEstimateColumnStatistics
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java:157]: newEstimateColumnStatistics.setNullsFraction() not set in else scope

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java [hunk 3]
```diff
@@ -119,15 +125,54 @@ public static Statistics estimateColumnToColumnComparison(
...
         double selectivity = 1.0 / Math.max(1, Math.max(leftDistinctValuesCount, rightDistinctValuesCount));
-        double rowCount = statistics.getOutputRowCount() * selectivity;
-        return Statistics.buildFrom(statistics).setOutputRowCount(rowCount).build();
+        double rowCount = statistics.getOutputRowCount() * selectivity *
+                (isEqualForNull ? 1 :
+                        (1 - leftColumnStatistic.getNullsFraction()) * (1 - rightColumnStatistic.getNullsFraction()));
+
+        StatisticRangeValues intersect = StatisticRangeValues.from(leftColumnStatistic)
+                .intersect(StatisticRangeValues.from(rightColumnStatistic));
+        ColumnStatistic.Builder newEstimateColumnStatistics = ColumnStatistic.builder().
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java:143]: AverageRowSize?

**Review by Youngwb — APPROVED** [review]

**Comment by kangkaisen** [comment]
Please create a issue

**Comment by HangyuanLiu** [comment]
> Please create a issue

ok

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**